### PR TITLE
FW-5622 Converted ImmersionLabels from controlled to uncontrolled content

### PR DESCRIPTION
**Pull request by sahil97**
### Description of Changes
Please briefly explain the changes here.

### Checklist
- [x] README / documentation has been updated if needed
- [x] PR title / commit messages contain Jira ticket number if applicable
- [x] Tests have been updated / created if applicable
- ~Admin Panel has been updated if models have been added or modified~
- [x] Migrations have been updated and committed if applicable
- ~Insomnia workspace has been updated if applicable~

### Reviewers Should Do The Following:
- [x] Review the code changes here
- ~Pull the branch and test locally~

### Additional Notes
N/A

### CODE REVIEW

**Review by bronson-c — APPROVED** [review]
LGTM! 👍

File: firstvoices/backend/models/immersion_labels.py [hunk 1]
```diff
@@ -22,10 +22,10 @@ class Meta:
...
         ]
         rules_permissions = {
-            "view": predicates.is_visible_object,
-            "add": predicates.is_language_admin_or_super,
-            "change": predicates.is_language_admin_or_super,
-            "delete": predicates.is_language_admin_or_super,
+            "view": predicates.has_visible_site,
+            "add": predicates.can_add_core_uncontrolled_data,
+            "change": predicates.can_edit_core_uncontrolled_data,
+            "delete": predicates.can_delete_core_uncontrolled_data,
```
Comment (sarahfirstvoices) [inline firstvoices/backend/models/immersion_labels.py:28]: This is a big permission change that I don't think we want to make. Immersion Labels are not "core" uncontrolled data-- that was meant to cover media.

Our roles and permissions doc lists Immersion Labels as only editable by Language Admins and Superadmins: https://firstvoices.atlassian.net/wiki/spaces/FIR/pages/220790795/User+Roles+in+fv-be

Comment (sarahfirstvoices) [reply firstvoices/backend/models/immersion_labels.py:28]: View permission looks good though

File: firstvoices/backend/views/immersion_label_views.py [hunk 2]
```diff
@@ -132,8 +134,13 @@ class ImmersionLabelViewSet(
 
     def get_queryset(self):
         site = self.get_validated_site()
+        visible_dictionary_entry = DictionaryEntry.objects.visible(
+            self.request.user
+        ).filter(immersion_labels=OuterRef("pk"))
```
Comment (sarahfirstvoices) [inline firstvoices/backend/views/immersion_label_views.py:139]: Another, somewhat simpler way to accomplish this is to use a condition when prefetching the dictionary entry. We do something similar for the site features in the Sites API (those are filtered by value not visibility but it shows the method).

**Review by sarahfirstvoices — CHANGES_REQUESTED** [review]

File: firstvoices/backend/permissions/predicates/base.py [hunk 3]
```diff
@@ -135,3 +135,27 @@ def is_hidden_site(user, obj):
...
+def has_related_public_dictionary_entry(user, obj):
+    return obj.dictionary_entry.visibility == Visibility.PUBLIC
+
+
+@predicate
+def has_related_team_dictionary_entry(user, obj):
+    return obj.dictionary_entry.visibility == Visibility.TEAM
+
+
+has_member_access_to_related_dictionary_entry = (
```
Comment (bronson-c) [inline firstvoices/backend/permissions/predicates/base.py:153]: If this predicate is meant to be used the same as `has_public_access_to_related_dictionary_entry` it is missing the `Predicate()` and also missing a name

**Review by bronson-c — APPROVED** [review]
Outside of a few additions required for the `has_member_access_to_related_dictionary_entry` predicate the changes LGTM!

File: firstvoices/backend/permissions/predicates/view_models.py [hunk 4]
```diff
@@ -24,3 +24,15 @@
     ),
     name="can_view_membership_model",
 )
+
+
+# Can view immersion label, if the related dictionary-entry should be public
+can_view_immersion_label = Predicate(
```
Comment (sarahfirstvoices) [inline firstvoices/backend/permissions/predicates/view_models.py:30]: For the predicate, I think we could avoid some duplication by doing

```
@predicate
def can_view_immersion_label(user, obj):
    return is_visible_object(user, obj.dictionary_entry)
```

File: firstvoices/backend/permissions/predicates/base.py (hunk 3 above)
Comment (sahil97) [reply firstvoices/backend/permissions/predicates/base.py]: Yes, those have been removed now with the reuse of `is_visible_object` predicate.

File: firstvoices/backend/tests/test_apis/test_parts_of_speech_api.py [hunk 5]
```diff
@@ -65,15 +65,12 @@ def test_detail_view(self, db):
         # test will fail if fixture not loaded since there will be no items without any children in the response to
         # choose the first element from. If fixture is modified in such as way that all elements now have children at
         # top level, Create another fixture for testing purposes with one such element.
-        item = None
-        for x in list_data:
-            if x["parent"]:
-                item = x
+        for item in list_data:
```
Comment (sahil97) [inline firstvoices/backend/tests/test_apis/test_parts_of_speech_api.py:68]: Unrelated to the PR. Fixed so sonar can pass our main branch build.

**Review by sarahfirstvoices — APPROVED** [review]
Thanks for making those changes, and adding the fix for the main branch!

**Pull request merged**